action.yaml

```diff
@@ -2,9 +2,17 @@ name: 'Aqua Security Trivy'
 description: 'Scans container images for vulnerabilities with Trivy'
 author: 'Aqua Security'
 inputs:
+  scan-type:
+    description: 'Scan type to use for scanning vulnerability'
+    required: false
+    default: 'image'
   image-ref:
-    description: 'image reference'
+    description: 'image reference(for backward compatibility.)'
     required: true
+  scan-ref:
+    description: 'Scan reference'
+    required: false
+    default: ${{ inputs.image-ref }}
```

Comment:
Does this actually work? Have you tried to run it?

Reply:
I just tried this via a test repo and it doesn't seem to work as I expected. 

<img width="811" alt="image" src="https://user-images.githubusercontent.com/1254783/107807988-c914e100-6d1d-11eb-8c5e-b300096b153c.png">

I suggest creating a test repo to try the changes here first. I would like to avoid the breaking change of changing the flag name if we can. Most people don't version their GitHub actions and might be using ours set to master as well rather than a released version.

Reply:
Thanks, @simar7 for the review, that was tricky since nowhere in the doc it is mentioned to add conditional flow for runs, I tested with 1 of the solution which is working and is backward compatible.
When scan-ref is passed in workflow [working](https://github.com/rahul2393/trivy-ci-test/runs/1893919460?check_suite_focus=true)
When image-ref is passed in workflow OR scan-ref is not set [working](https://github.com/rahul2393/trivy-ci-test/runs/1893909091?check_suite_focus=true)